python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -224,18 +222,6 @@ def __init__(
             backfill_policy, "backfill_policy", BackfillPolicy
         )
 
-        if self._partitions_def is None:
```

Comment:
This check was added as part of an external contribution and doesn't really make sense

Reply:
Ah I was wondering about this one. I saw this as part of my work in this area of the code base and it didn't make much sense to me.